src/main/kotlin/no/skatteetaten/aurora/gobo/graphql/environment/EnvironmentStatusDataLoader.kt

```diff
@@ -3,28 +3,89 @@ package no.skatteetaten.aurora.gobo.graphql.environment
 import no.skatteetaten.aurora.gobo.graphql.GoboDataLoader
 import no.skatteetaten.aurora.gobo.graphql.GoboGraphQLContext
 import no.skatteetaten.aurora.gobo.graphql.applicationdeployment.ApplicationDeploymentRef
+import no.skatteetaten.aurora.gobo.graphql.environment.EnvironmentStatusType.APPLIED
+import no.skatteetaten.aurora.gobo.graphql.environment.EnvironmentStatusType.INACTIVE
+import no.skatteetaten.aurora.gobo.integration.mokey.ApplicationDeploymentResource
 import no.skatteetaten.aurora.gobo.integration.mokey.ApplicationService
+import no.skatteetaten.aurora.gobo.integration.phil.DeploymentRefInput
+import no.skatteetaten.aurora.gobo.integration.phil.DeploymentResource
+import no.skatteetaten.aurora.gobo.integration.phil.DeploymentStatus
+import no.skatteetaten.aurora.gobo.integration.phil.EnvironmentService
 import org.springframework.stereotype.Component
 
-// TODO filter of statuses from Phil
-// 1) If status from Phil is failed, use it
-// 2) If status from Phil is success, use status from ApplicationDeployment (Mokey)
 @Component
-class EnvironmentStatusDataLoader(private val applicationService: ApplicationService) :
-    GoboDataLoader<EnvironmentApplication, EnvironmentStatus>() {
-
-    override suspend fun getByKeys(keys: Set<EnvironmentApplication>, context: GoboGraphQLContext):
-        Map<EnvironmentApplication, EnvironmentStatus> {
-        val applicationDeployments = applicationService.getApplicationDeployments(
-            keys.map { ApplicationDeploymentRef(it.environment, it.name) }
-        )
-
-        return keys.map { app ->
-            val ad =
-                applicationDeployments.find { ad ->
-                    ad.affiliation == app.affiliation && ad.environment == app.environment && ad.name == app.name
+class EnvironmentStatusDataLoader(
```

Comment:
Ingen krise akkurat, men føler at denne dataloaderen begynner å inneholde litt for mye. Er det funksjonalitet her vi kan flytte ut?

Reply:
Skal se på det